tests/all-test.php

```diff
@@ -591,4 +591,87 @@ public function test_user_logged_in_multisite() {
 		$output = ob_get_clean();
 		self::assertContains( self::$parsely_html, $output );
 	}
+
+	/**
+	 * Data provider for test_get_current_url
+	 *
+	 * @return array[]
+	 */
+	public function data_for_test_get_current_url() {
+		return array(
+			// Start cases with 'force_https_canonicals' = true
+			array(
+				true,
+				'http://example.com',
+				'https://example.com',
+			),
+			array(
+				true,
+				'https://example.com',
+				'https://example.com',
+			),
+			array(
+				true,
+				'http://example.com:1234',
+				'https://example.com:1234',
+			),
+			array(
+				true,
+				'https://example.com:1234',
+				'https://example.com:1234',
+			),
+			// Start cases with 'force_https_canonicals' = false
+			array(
+				false,
+				'http://example.com',
+				'http://example.com',
+			),
+			array(
+				false,
+				'https://example.com',
+				'http://example.com',
+			),
+			array(
+				false,
+				'http://example.com:1234',
+				'http://example.com:1234',
+			),
+			array(
+				false,
+				'https://example.com:1234',
+				'http://example.com:1234',
+			),
```

Comment:
How about mixing some of these up to test other strings, like `http://localhost:999` or `https://my-domain.local:1000/foo/bar`, since you're updating the site URL anyway?